spec/controllers/ops_controller/settings/upload_spec.rb

```diff
@@ -3,6 +3,166 @@
     stub_user(:features => :all)
   end
 
+  describe '#upload_png' do
+    let(:dir) { File.dirname(__FILE__) }
+
+    before do
+      allow(controller).to receive(:load_edit).and_return(true)
+      allow(controller).to receive(:redirect_to)
+      controller.instance_variable_set(:@sb, :active_tab => 'settings_custom_logos', :selected_server_id => 123)
+    end
+
+    it 'adds success flash message to @flash_array regarding updating custom logo image with a .png file' do
+      file = dir << '/data/test.png'
+      temp = Tempfile.new(['RackMultipart', '.png'])
+      png = File.open(file)
+      encoded_png = Base64.encode64(File.read(png.path))
+      File.open(temp.path, "wb") do |f|
+        f.write(Base64.decode64(encoded_png))
+      end
+      controller.params = {:upload => {:logo => ActionDispatch::Http::UploadedFile.new(:tempfile => temp, :filename => 'test.png', :type => 'image/png')}}
+      controller.send(:upload_logo)
+      expect(controller.instance_variable_get(:@flash_array)).to include(:message => 'Custom logo image "test.png" uploaded', :level => :success)
+    end
+
+    it 'adds error flash message to @flash_array regarding updating custom logo image with a non-png file' do
+      file = dir << '/data/test.bat.png'
+      temp = Tempfile.new(['RackMultipart', '.png'])
+      png = File.open(file)
+      encoded_png = Base64.encode64(File.read(png.path))
+      File.open(temp.path, "wb") do |f|
+        f.write(Base64.decode64(encoded_png))
+      end
+      controller.params = {:upload => {:logo => ActionDispatch::Http::UploadedFile.new(:tempfile => temp, :filename => 'test.bat.png', :type => 'image/png')}}
+      controller.send(:upload_logo)
+      expect(controller.instance_variable_get(:@flash_array)).to include(:message => 'Custom logo image must be a .png file', :level => :error)
+
+      dir = File.dirname(__FILE__)
+      file = dir << '/data/test.ico'
```

Comment:
This says test.ico, but the filename on line 48 says "test.bat.png".  It also seems copy-pasted down in the other png methods as well?

Reply:
Yeah I think I just copy/pasted and forgot to change it.